src/libraries/System.Drawing.Common/src/System/Drawing/DrawingComWrappers.cs

```diff
@@ -0,0 +1,331 @@
+            {
+                try
+                {
+                    Interop.Ole32.IStream inst = ComInterfaceDispatch.GetInstance<Interop.Ole32.IStream>((ComInterfaceDispatch*)thisPtr);
+                    Interop.Ole32.IStream pstmStream = ComInterfaceDispatch.GetInstance<Interop.Ole32.IStream>((ComInterfaceDispatch*)pstm);
+
+                    inst.CopyTo(pstmStream, cb, pcbRead, pcbWritten);
+                }
+                catch (Exception e)
+                {
+                    return e.HResult;
+                }
+
+                return OK;
+            }
+
+            [UnmanagedCallersOnly]
+            private static int CommitImplementation(IntPtr thisPtr, uint grfCommitFlags)
+            {
+                try
+                {
+                    Interop.Ole32.IStream inst = ComInterfaceDispatch.GetInstance<Interop.Ole32.IStream>((ComInterfaceDispatch*)thisPtr);
+                    inst.Commit(grfCommitFlags);
+                }
+                catch (Exception e)
+                {
+                    return e.HResult;
+                }
+
+
+                return OK;
+            }
+
+            [UnmanagedCallersOnly]
+            private static int RevertImplementation(IntPtr thisPtr)
+            {
+                try
+                {
+                    Interop.Ole32.IStream inst = ComInterfaceDispatch.GetInstance<Interop.Ole32.IStream>((ComInterfaceDispatch*)thisPtr);
+                    inst.Revert();
+                }
+                catch (Exception e)
+                {
+                    return e.HResult;
+                }
+
+
+                return OK;
+            }
+
+            [UnmanagedCallersOnly]
+            private static int LockRegionImplementation(IntPtr thisPtr, ulong libOffset, ulong cb, uint dwLockType)
+            {
+                try
+                {
+                    Interop.Ole32.IStream inst = ComInterfaceDispatch.GetInstance<Interop.Ole32.IStream>((ComInterfaceDispatch*)thisPtr);
+                    return (int)inst.LockRegion(libOffset, cb, dwLockType);
+                }
+                catch (Exception e)
+                {
+                    return e.HResult;
+                }
+            }
+
+            [UnmanagedCallersOnly]
+            private static int UnlockRegionImplementation(IntPtr thisPtr, ulong libOffset, ulong cb, uint dwLockType)
+            {
+                try
+                {
+                    Interop.Ole32.IStream inst = ComInterfaceDispatch.GetInstance<Interop.Ole32.IStream>((ComInterfaceDispatch*)thisPtr);
+                    return (int)inst.UnlockRegion(libOffset, cb, dwLockType);
+                }
+                catch (Exception e)
+                {
+                    return e.HResult;
+                }
+            }
+
+            [UnmanagedCallersOnly]
+            private static int StatImplementation(IntPtr thisPtr, out Interop.Ole32.STATSTG pstatstg, Interop.Ole32.STATFLAG grfStatFlag)
```

Comment:
I would convert `out Interop.Ole32.STATSTG` to `Interop.Ole32.STATSTG* `

Reply:
I'm pretty sure this change is required for this to work.

Reply:
I thought so too. I am wondering if the failure mode is an assert only or if this simply doesn't get called in the current scenario.

Reply:
I believe the failure mode is an exception, but only at function call time.

Reply:
I believe in the one scenario I am converting in this PR, this method isn't called. Only the `Write` method gets called, which is why I wasn't seeing a failure.

Fixing.